pkg/daemon/daemon.go

```diff
@@ -124,13 +140,21 @@ func New(
 		}
 		glog.Infof("Booted osImageURL: %s (%s)", osImageURL, osVersion)
 	}
+
+	currentBootIDBytes, err := ioutil.ReadFile("/proc/sys/kernel/random/boot_id")
+	if err != nil {
+		return nil, err
+	}
+	currentBootID := strings.TrimSpace(string(currentBootIDBytes))
+
 	dn := &Daemon{
 		name:                   nodeName,
 		OperatingSystem:        operatingSystem,
 		NodeUpdaterClient:      nodeUpdaterClient,
 		loginClient:            loginClient,
 		rootMount:              rootMount,
 		fileSystemClient:       fileSystemClient,
+		bootID:                 currentBootID,
```

Comment:
Does the `NewClusterDrivenDaemon` need to know the `bootID` or only when it's not cluster connected?

Reply:
Currently we're assigning it in both cases (this is the common constructor).  However whether that's necessary gets into the onceFrom question.